[export][refactor][4/n] Make equality_constraints optional

angelayi opened the pull request:
Summary: needed to remove equality_contraints eventually :P

Test Plan: CI

Differential Revision: D52351709

zhxchen17 commented on torch/export/exported_program.py:
```diff
@@ -122,10 +122,10 @@ def __init__(
+        self._equality_constraints: List[Tuple[InputDim, InputDim]] = (
+            equality_constraints or []
+        )
+        self._module_call_graph: List[ModuleCallEntry] = module_call_graph or []
```
Suggestion: If it's too hard to make it not optional for now, better enforce the invariants in code directly.
```suggestion
        assert module_call_graph is not None
        self._module_call_graph: List[ModuleCallEntry] = module_call_graph
        assert len(self._module_call_graph) >= 1
```

zhxchen17 reviewed (changes requested):
not addressed yet. send back to your queue.

angelayi committed:
[export][refactor][4/n] Make equality_constraints optional and remove OSS uses (#116233)

Summary:

needed to remove equality_contraints eventually :P

Test Plan: CI

Reviewed By: tugsbayasgalan

Differential Revision: D52351709
